### ISSUE TITLE
Might not be able to port to Homebrew yet

### ISSUE THREAD

**Issue by gouravkhunger** [srevinsaju/guiscrcpy#279]
Hey!
I tried to make a [Homebrew tap](https://github.com/gouravkhunger/homebrew-guiscrcpy) which could potentially allow m1 users to install and run guiscrcpy by running:

```
brew tap gouravkhunger/guiscrcpy && brew install guiscrcpy
```

But unfortunately, it seems that taps which have commands that require x86 architecture can't be run through m1 version of Homebrew even with rosetta emulation.

<img width="545" alt="image" src="https://user-images.githubusercontent.com/46792249/145071176-f10bf8cd-baa1-46c1-9eb1-82bb34e2b8e7.png">

If the same command is run in arm based terminal, it gives this error.

<img width="581" alt="image" src="https://user-images.githubusercontent.com/46792249/145071306-b11a7f5b-0e0c-4e96-9169-5370cc6abf01.png">

And if this gets fixed, then `shiboken2` package fails installing because of the same `python3.9` conflict. Also, seemingly there is no way to unlink python3.9 even by changing poetry environment unless the user explicitly unlinks through a brew command. The Homebrew version can be ported only when `QtPy` and `PySide2` get fully m1 compatible.

Till then, I have made a small bash script that can do the [tedious work](https://github.com/srevinsaju/guiscrcpy/blob/main/docs/BUILD.md#m1-specific-build-instructions) needed to build `guiscrcpy`. If you wish I can make a pull request to the same in the installation instructions!

Btw, I figured out that we don't need to uninstall `python3.9` to be able to run it. A simple poetry command can let us keep all python versions and use the one that works (python 3.8 here).

```
poetry env use /usr/bin/python3
```

This forces the use of default python 3.8 rather than Homebrew installed python 3.9/3.10 which earlier caused issues. I can update the docs in the same PR 👍🏻

**Comment by srevinsaju** [comment]
> Hey!
> I tried to make a Homebrew tap which could potentially allow m1 users to install and run guiscrcpy by running:

Thanks, looks interesting, and we are probably heading in the right direction!.

How about we try this method: Using PyInstaller to create ready to install binaries of `guiscrcpy` for macOS? This will ensure that guiscrcpy has a binary of python included in itself, and we will not need to depend on python from homebrew. What do you say?

 https://pyinstaller.readthedocs.io/en/stable/usage.html#mac-os-specific-options

**Comment by gouravkhunger** [comment]
Yes, we could package dependencies and provide a standalone app. Will look further into this 👍🏻

**Comment by Genxster1998** [comment]
Is it going anywhere ? It sucks to download 3gb+ homebrew repo just for this , another python based package manager will be great for this .
